public/app/core/components/Login/LoginPage.test.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { LoginPage } from './LoginPage';
+
+const postMock = jest.fn();
+jest.mock('@grafana/runtime', () => ({
+  getBackendSrv: () => ({
+    post: postMock,
+  }),
+}));
+
+jest.mock('app/core/config', () => {
+  return {
+    loginError: false,
+    buildInfo: {
+      version: 'v1.0',
+      commit: '1',
+      env: 'production',
+      edition: 'Open Source',
+      isEnterprise: false,
+    },
+    licenseInfo: {
+      stateInfo: '',
+      licenseUrl: '',
+    },
+    appSubUrl: '',
+    getConfig: () => ({
+      appSubUrl: '',
+      verifyEmailEnabled: false,
+    }),
+  };
+});
+
+describe('Login Page', () => {
+  it('renders correctly', () => {
+    render(<LoginPage />);
+
+    expect(screen.getByText('Welcome to Grafana')).toBeInTheDocument();
+    expect(screen.getByLabelText('Username input field')).toBeInTheDocument();
+    expect(screen.getByLabelText('Password input field')).toBeInTheDocument();
+    expect(screen.getByLabelText('Login button')).toBeInTheDocument();
+
+    expect(screen.getByText('Forgot your password?')).toBeInTheDocument();
+    expect(screen.getByText('Forgot your password?').closest('a')).toHaveAttribute(
+      'href',
+      '/user/password/send-reset-email'
+    );
+
+    expect(screen.getByText('Sign up')).toBeInTheDocument();
+    expect(screen.getByText('Sign up').closest('a')).toHaveAttribute('href', '/signup');
+  });
```

Comment:
```suggestion
describe('Login Page', () => {
  it('renders correctly', () => {
    render(<LoginPage />);

    expect(screen.getByRole('heading', { name: 'Welcome to Grafana' })).toBeInTheDocument();
    expect(screen.getByRole('textbox', { name: 'Username input field' })).toBeInTheDocument();
    expect(screen.getByLabelText('Password input field')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Login button' })).toBeInTheDocument();

    expect(screen.getByRole('link', { name: 'Forgot your password?' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Forgot your password?' })).toHaveAttribute(
      'href',
      '/user/password/send-reset-email'
    );

    expect(screen.getByRole('link', { name: 'Sign up' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Sign up' })).toHaveAttribute('href', '/signup');
  });
```

Same comment about using `*ByRole` query here.

Reply:
ok. addressed the above changes.